routing.go

```diff
@@ -270,6 +270,20 @@ func (dht *IpfsDHT) GetValues(ctx context.Context, key string, nvals int) (_ []R
 	return out, ctx.Err()
 }
 
+// Computes the desired number of results for each path. This is roughly
+// resultsWanted / numPaths, but taking into account the index allows
+// computing the exact size for each path such that the sum is resultsWanted.
+func pathSize(resultsWanted int, pathIndex int, numPaths int) int {
+	size := resultsWanted / numPaths
```

Comment:
`resultsWanted` will be `-1` for `searchValue` when quorum size is not specified (used to indicate 'search as long as you have peers to query and didn't run out of time'), this might be wrong.

Reply:
Good catch, it's definitely wrong!

Reply:
Is there a test for this?

Reply:
There is one test that should show this:

https://github.com/libp2p/go-libp2p-kad-dht/blob/94794a5087c39d7e281f37c8ccad76c416ec6745/dht_test.go#L331

Reply:
Presumably this only applies to `searchValue`, not `FindProvidersAsync`, right? I can't see how `FindProvidersAsync` would work correctly with `count == -1` on master.